examples/systemcalls/systemcalls.c

```diff
@@ -92,6 +150,56 @@ bool do_exec_redirect(const char *outputfile, int count, ...)
  *   The rest of the behaviour is same as do_exec()
  *
 */
+    int fd = open(outputfile, (O_RDWR | O_CREAT | O_TRUNC),
```

Comment:
@ameyflash think about how you could avoid duplicating the logic from the function above in this function.